homeassistant/components/bmw_connected_drive/lock.py

```diff
@@ -59,10 +64,14 @@ def name(self):
     def device_state_attributes(self):
         """Return the state attributes of the lock."""
         vehicle_state = self._vehicle.state
-        return {
+        result = {
             "car": self._vehicle.name,
-            "door_lock_state": vehicle_state.door_lock_state.value,
+            ATTR_ATTRIBUTION: ATTRIBUTION,
         }
+        if self.door_lock_state_available:
+            result["door_lock_state"] = vehicle_state.door_lock_state.value
+            result["last_update_reason"] = vehicle_state.last_update_reason
+        return sorted(result.items())
```

Comment:
Can you just return `result` here?

Reply:
I have used sorting here just as in binary sensor. But can be changed of course, let me know what is prefered.

Reply:
While Python 3.7 does have ordered dictionaries, homeassistant.helpers.entity._async_write_ha_state combines state_attributes and device_state_attributes so there isn't a guarantee they will stay ordered when combined or that won't change in the future.

Reply:
Only the device_state_attributes are used here and HA returns the list sorted right now. So I would prefer to keep it like this.

![Screenshot_3](https://user-images.githubusercontent.com/11230573/76266832-17497680-6269-11ea-85d7-0ac511f85e7c.jpg)

Reply:
We shouldn't change the return type. This property should return a dictionary. Please make sure that we do that.

Reply:
Changed it to directly return the result.